src/app/agent/agent-dashboard/select-traveler/select-traveler.component.ts

```diff
@@ -1,39 +1,61 @@
-import { Component, OnInit, EventEmitter, Output } from '@angular/core';
-import { FormGroup, FormControl, Validators, FormBuilder } from '@angular/forms';
-import { environment } from 'src/environments/environment';
-import { Traveler } from '../../../common/entities/Traveler';
-import { AgentUtopiaService } from 'src/app/common/h/agent-utopia.service';
+import { Component, OnInit, EventEmitter, Output } from "@angular/core";
+import {
+  FormGroup,
+  FormControl,
+  Validators,
+  FormBuilder,
+  Form,
+} from "@angular/forms";
+import { environment } from "src/environments/environment";
+import { Traveler } from "../../../common/entities/Traveler";
+import { AgentUtopiaService } from "src/app/common/h/agent-utopia.service";
 
 @Component({
-  selector: 'app-agent-select-traveler',
-  templateUrl: './select-traveler.component.html',
-  styleUrls: ['./select-traveler.component.css']
+  selector: "app-agent-select-traveler",
+  templateUrl: "./select-traveler.component.html",
+  styleUrls: ["./select-traveler.component.css"],
 })
 export class SelectTravelerComponent implements OnInit {
-  selectTravelerForm = new FormGroup({
-    username: new FormControl('', [<any>Validators.required, <any>Validators.minLength(5)]),
-  });
-
-  createTravelerForm = new FormGroup({
-    name: new FormControl('', [<any>Validators.required, <any>Validators.minLength(5)]),
-    username: new FormControl('', [<any>Validators.required, <any>Validators.minLength(5)]),
-    email: new FormControl('', [<any>Validators.required, <any>Validators.minLength(5)]),
-  });
-
+  selectTravelerForm: FormGroup;
+  createTravelerForm: FormGroup;
   traveler: Traveler;
   invalidLogin: boolean;
   createTraveler: boolean;
   usernameTaken: boolean;
 
   @Output() travelerChanged = new EventEmitter<Traveler>();
+  invalidLoginSpace: boolean;
 
-  constructor(private service: AgentUtopiaService,
-    private formBuilder: FormBuilder,) {
-    this.createSelectTravelerForm();
-   }
+  constructor(
+    private service: AgentUtopiaService,
+    private formBuilder: FormBuilder
+  ) {
+  }
 
   ngOnInit() {
     this.createTraveler = false;
+    
+    this.selectTravelerForm = new FormGroup({
+      username: new FormControl(null, [
+        Validators.required,
+        Validators.maxLength(45),
+      ]),
+    });
+
+    this.createTravelerForm = new FormGroup({
+      name: new FormControl(null, [
+        Validators.required,
+        Validators.maxLength(45),
+      ]),
+      username: new FormControl(null, [
+        Validators.required,
+        Validators.maxLength(45),
+      ]),
+      email: new FormControl(null, [
+        Validators.required,
+        Validators.maxLength(45),
+      ]),
```

Comment:
The max length for this should maybe be longer, since email addresses can be fairly long. For now, since it's not going into the database, we don't strictly need a max length at all. Also, it would be good to have a validator that makes sure the input matches the format of an email address. I think Angular has one built in.